Framework/PythonInterface/mantid/api/src/Exports/AnalysisDataServiceObserver.cpp

```diff
@@ -7,34 +7,76 @@
+      .def("observeReplace",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeReplace),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a workspace being replaced")
-      .def("observeDelete", &AnalysisDataServiceObserverAdapter::observeDelete, (arg("self"), arg("on")),
+      .def("observeDelete",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeDelete),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a workspace being deleted")
-      .def("observeClear", &AnalysisDataServiceObserverAdapter::observeClear, (arg("self"), arg("on")),
+      .def("observeClear",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeClear),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for it being cleared")
-      .def("observeRename", &AnalysisDataServiceObserverAdapter::observeRename, (arg("self"), arg("on")),
+      .def("observeRename",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeRename),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a workspace being renamed")
-      .def("observeGroup", &AnalysisDataServiceObserverAdapter::observeGroup, (arg("self"), arg("on")),
+      .def("observeGroup",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeGroup),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a group being added/made in the ADS")
-      .def("observeUnGroup", &AnalysisDataServiceObserverAdapter::observeUnGroup, (arg("self"), arg("on")),
+      .def("observeUnGroup",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeUnGroup),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a group being removed from the ADS")
-      .def("observeGroupUpdate", &AnalysisDataServiceObserverAdapter::observeGroupUpdate, (arg("self"), arg("on")),
+      .def("observeGroupUpdate",
+           make_function(std::bind(callReleasingGIL, _1, _2, &AnalysisDataServiceObserver::observeGroupUpdate),
+                         default_call_policies(), (arg("self"), arg("on")), ObserverMethodSignature()),
            "Observe AnalysisDataService for a group being updated by being "
            "added to or removed from");
 }
```

Comment:
Is there a missing newline here?

Reply:
Good catch. Sounds like something we should add to the pre-commit checks.